### PR TITLE
fix: manual fix of eslint errors on packages @o3r/{d,e,f}*

### PR DESCRIPTION
**Pull request by sdo-1A**
## Proposed change

Manual fix of ESLint errors for @o3r/design, @o3r/dev-tools, @o3r/dynamic-content, @o3r/eslint-config, @o3r/eslint-config-otter, @o3r/extractors, @o3r/forms

## Related issues

- :bug: Fixes #(issue)
- :rocket: Feature #(issue)

### CODE REVIEW

File: packages/@o3r/design/builders/generate-css/index.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ import type {
 } from './schema';
 
 const createBuilderWithMetricsIfInstalled: BuilderWrapper = (builderFn) => async (opts, ctx) => {
+  // eslint-disable-next-line unicorn/consistent-function-scoping -- keep for readability
   let wrapper: BuilderWrapper = (fn) => fn;
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/builders/generate-css/index.ts:48]:
```suggestion
const noopBuilder: BuilderWrapper = (fn) => fn;

const createBuilderWithMetricsIfInstalled: BuilderWrapper = (builderFn) => async (opts, ctx) => {
  let wrapper: BuilderWrapper = noopBuilder;
```

File: packages/@o3r/design/schematics/extract-token/index.ts [hunk 2]
```diff
@@ -46,7 +46,7 @@ function extractTokenFn(options: ExtractTokenSchematicsSchema): Rule {
 
   return async (tree, context) => {
     try {
-      // eslint-disable-next-line @typescript-eslint/naming-convention
+      // eslint-disable-next-line @typescript-eslint/naming-convention -- convention for class name
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/schematics/extract-token/index.ts:49]: I would put something like `false positive when using dynamic import` cause we deactivate this rule for imports
and we should check if an issue is already opened or if we should open one :)

File: packages/@o3r/design/schematics/ng-add/index.ts [hunk 3]
```diff
@@ -27,15 +27,15 @@ export function ngAddFn(options: NgAddSchematicsSchema): Rule {
     return chain([
       registerGenerateCssBuilder(options.projectName),
       setupSchematicsParamsForProject({
-        // eslint-disable-next-line @typescript-eslint/naming-convention
+        // eslint-disable-next-line @typescript-eslint/naming-convention -- convention for name of schematic in angular.json
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/schematics/ng-add/index.ts:30]: Could be fixed with 
```ts
const defaultSchematicOptions = Object.fromEntries([
  '@o3r/core:component',
  '@o3r/core:component-presenter',
  '*:*'
].map((name) => [name, { useOtterDesignToken: true }]);
...
setupSchematicsParamsForProject(
  defaultSchematicOptions,
  options.projectName
)

File: packages/@o3r/design/src/core/design-token/design-token-specification.interface.ts [hunk 4]
```diff
@@ -290,8 +288,7 @@ export const isDesignTokenGroup = (node?: any): node is DesignTokenGroup => {
  * @param value Stroke Style value
  * @returns true if it is a defined value
  */
-// eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
-export const isTokenTypeStrokeStyleValueComplex = (value?: DesignTokenTypeStrokeStyleValue | string): value is DesignTokenTypeStrokeStyleDetailsValue => {
+export const isTokenTypeStrokeStyleValueComplex = (value?: DesignTokenTypeStrokeStyleValue): value is DesignTokenTypeStrokeStyleDetailsValue => {
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/src/core/design-token/design-token-specification.interface.ts:291]:
```suggestion
export const isTokenTypeStrokeStyleValueComplex = (value?: Exclude<DesignTokenTypeStrokeStyleValue, string> | string): value is DesignTokenTypeStrokeStyleDetailsValue => {
```
else we could not call this method `isTokenTypeStrokeStyleValueComplex('my-custom-string-not-matching-string-in-DesignTokenTypeStrokeStyleValue')

File: packages/@o3r/design/src/core/design-token/renderers/css/design-token-value.renderers.ts [hunk 5]
```diff
@@ -56,12 +56,14 @@ export const getCssTokenValueRenderer = (options?: CssTokenValueRendererOptions)
...
-    if (isPrivateVariable(variable)) {
-      // eslint-disable-next-line no-use-before-define
-      return `var(--${variable.getKey(tokenVariableNameRenderer)}, ${renderer(variable, variableSet)})`;
-    } else {
-      return `var(--${variable.getKey(tokenVariableNameRenderer)})`;
-    }
+    return isPrivateVariable(variable)
+      ? `var(--${variable.getKey(tokenVariableNameRenderer)}, ${renderer(variable, variableSet)})`
+      : `var(--${variable.getKey(tokenVariableNameRenderer)})`;
+    // if (isPrivateVariable(variable)) {
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/src/core/design-token/renderers/css/design-token-value.renderers.ts:62]: comment could be removed :)

File: packages/@o3r/design/src/core/design-token/renderers/design-token/design-token-value.renderers.ts [hunk 6]
```diff
@@ -15,6 +15,7 @@ export interface DesignTokenTokenValueRendererOptions {
  * @param _options
  */
 export const getDesignTokenTokenValueRenderer = (_options?: DesignTokenTokenValueRendererOptions): TokenValueRenderer => {
+  // eslint-disable-next-line unicorn/consistent-function-scoping -- keep for readability
   const renderer = (variable: DesignTokenVariableStructure, _variableSet: Map<string, DesignTokenVariableStructure>) => {
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/src/core/design-token/renderers/design-token/design-token-value.renderers.ts:19]: could be extracted

File: packages/@o3r/design/src/core/design-token/renderers/sass/design-token-definition.renderers.spec.ts [hunk 7]
```diff
@@ -40,6 +40,7 @@ describe('getSassTokenDefinitionRenderer', () => {
   });
 
   test('should prefix private variable', () => {
+    // eslint-disable-next-line unicorn/consistent-function-scoping -- keep for readability
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/src/core/design-token/renderers/sass/design-token-definition.renderers.spec.ts:43]:
```suggestion
    // eslint-disable-next-line unicorn/consistent-function-scoping -- linked to this test only
```

File: packages/@o3r/design/src/core/design-token/renderers/sass/design-token-definition.renderers.ts [hunk 8]
```diff
@@ -42,7 +42,7 @@ export const tokenVariableNameSassRenderer: TokenKeyRenderer = (variable) => {
 /**
  * Retrieve the Design Token Variable renderer for Sass
  * @param options
- * @returns
+ * @returns tokenDefinitionRenderer
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/src/core/design-token/renderers/sass/design-token-definition.renderers.ts:45]:
```suggestion
```

File: packages/@o3r/dev-tools/src/cli/aql-artifact-cleaner.ts [hunk 9]
```diff
@@ -64,14 +64,15 @@ if (!opts.basicAuth && !opts.apiKey) {
   process.exit(1);
 }
 let authHeader: Headers;
```
Comment (matthieu-crouzet) [inline packages/@o3r/dev-tools/src/cli/aql-artifact-cleaner.ts:66]:
```ts
const authHeader: Headers = {
  ...(
   // eslint-disable-next-line @typescript-eslint/naming-convention -- naming convention required by HTTP
   opts.basicAuth ? {Authorization: 'Basic ' + (opts.basicAuth as string)} : {'X-JFrog-Art-Api': (opts.apiKey as string)}
  )
};
```
maybe a question of preference 
but I think ternary are easier to read 
plus in this case we can change `authHeader` into a `const` so it brings more information

File: packages/@o3r/dev-tools/src/cli/checks/in-source-dep-check.ts [hunk 10]
```diff
@@ -18,7 +18,7 @@ import {
   program
 } from 'commander';
 import * as glob from 'globby';
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment -- keep ts-comment for explanation
 // @ts-ignore -- _builtinLibs is not part of repl types (due to the fact it is flagged to internal usage purpose)
```
Comment (matthieu-crouzet) [inline packages/@o3r/dev-tools/src/cli/checks/in-source-dep-check.ts:22]:
```suggestion
// eslint-disable-next-line @typescript-eslint/ban-ts-comment -- _builtinLibs is not part of repl types (due to the fact it is flagged to internal usage purpose)
// @ts-ignore
```

File: packages/@o3r/dev-tools/src/cli/generate-package-exports.ts [hunk 11]
```diff
@@ -58,7 +58,7 @@ program
 
 logger.warn('This script is deprecated, will be removed in Otter v12.');
 
-// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion -- keep to assign type to variables (otherwise type `any`)
 const { cwd, pattern, exportTypes, ...options } = program.opts() as Options;
```
Comment (matthieu-crouzet) [inline packages/@o3r/dev-tools/src/cli/generate-package-exports.ts:62]:
```suggestion
const { cwd, pattern, exportTypes, ...options } = program.opts<Options>();
```

File: packages/@o3r/dev-tools/src/cli/version-harmonize.ts [hunk 12]
```diff
@@ -53,7 +53,7 @@ program
   .parse(process.argv);
 
 /** Options from CLI */
-// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion -- keep to assign type to variables (otherwise type `any`)
 const options = program.opts() as Options;
```
Comment (matthieu-crouzet) [inline packages/@o3r/dev-tools/src/cli/version-harmonize.ts:57]:
```suggestion
const options = program.opts<Options>();
```

File: packages/@o3r/dev-tools/src/utils/jest.helpers.ts [hunk 13]
```diff
@@ -54,14 +54,15 @@ const findParentPackageJson = (directory: string, rootDir?: string): string | un
  * @deprecated Please use `pathsToModuleNameMapper` from `ts-jest`, will be removed in Otter v12.
  * @param rootDir Root directory of the jest project
  * @param testingTsconfigPath Path to the tsconfig.json used for test mapping files
- * @returns
+ * @returns {Record<string, any>} List of modules mapping
```
Comment (matthieu-crouzet) [inline packages/@o3r/dev-tools/src/utils/jest.helpers.ts:57]:
```suggestion
```
not needed in typescript files as we already have the typings and the description of the function

File: packages/@o3r/dynamic-content/middlewares/parameter-extractor.middleware.ts [hunk 14]
```diff
@@ -1,6 +1,6 @@
 import * as querystring from 'node:querystring';
 import * as cheerio from 'cheerio';
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment -- keep ts-comment for explanation
 // @ts-ignore -- no type are provided for express-interceptor
```
Comment (matthieu-crouzet) [inline packages/@o3r/dynamic-content/middlewares/parameter-extractor.middleware.ts:4]:
```suggestion
// eslint-disable-next-line @typescript-eslint/ban-ts-comment -- no type are provided for express-interceptor
// @ts-ignore
```

File: packages/@o3r/dynamic-content/src/services/request-parameters/request-parameters.config.ts [hunk 15]
```diff
@@ -7,13 +7,13 @@
  */
 
 export enum StorageStrategy {
-  // eslint-disable-next-line @typescript-eslint/naming-convention
+  // eslint-disable-next-line @typescript-eslint/naming-convention -- required naming convention
```
Comment (matthieu-crouzet) [inline packages/@o3r/dynamic-content/src/services/request-parameters/request-parameters.config.ts:10]: We may discuss the configuration of this rule for `enum`

File: packages/@o3r/dynamic-content/src/services/request-parameters/request-parameters.spec.ts [hunk 16]
```diff
@@ -506,6 +506,7 @@ describe('RequestParametersService', () => {
 
   describe('No storage available', () => {
     beforeEach(async () => {
+      // eslint-disable-next-line unicorn/consistent-function-scoping -- keep for readability
```
Comment (matthieu-crouzet) [inline packages/@o3r/dynamic-content/src/services/request-parameters/request-parameters.spec.ts:509]:
```suggestion
      // eslint-disable-next-line unicorn/consistent-function-scoping -- used only in this scope
```

File: packages/@o3r/eslint-config-otter/schematics/ng-add/index.ts [hunk 17]
```diff
@@ -55,7 +55,7 @@ function ngAddFn(options: NgAddSchematicsSchema): Rule {
     const depsInfo = getO3rPeerDeps(path.resolve(__dirname, '..', '..', 'package.json'), true, /^@(?:o3r|ama-sdk|eslint-)/);
     const workspaceProject = options.projectName ? getWorkspaceConfig(tree)?.projects[options.projectName] : undefined;
     const linterSchematicsFolder = __dirname;
-    // eslint-disable-next-line @typescript-eslint/naming-convention
+    // eslint-disable-next-line @typescript-eslint/naming-convention -- convention for class name
```
Comment (matthieu-crouzet) [inline packages/@o3r/eslint-config-otter/schematics/ng-add/index.ts:58]:
```suggestion
    // eslint-disable-next-line @typescript-eslint/naming-convention -- false positive when using dynamic import
```

File: packages/@o3r/eslint-config-otter/testing/setup-jest.builders.ts [hunk 18]
```diff
@@ -0,0 +1 @@
+/* eslint-disable unicorn/no-empty-file -- will be deprecated in v12 */
```
Comment (matthieu-crouzet) [inline packages/@o3r/eslint-config-otter/testing/setup-jest.builders.ts:1]: we may configure this rule globally to not check `setup-jest*.ts`
as we have a global setup relying on this file even if not needed for this packge

File: packages/@o3r/extractors/schematics/cms-adapter/index.ts [hunk 19]
```diff
@@ -62,6 +63,7 @@ export function updateCmsAdapter(options: { projectName?: string | undefined },
     return rule(tree, context);
   };
 
+  // eslint-disable-next-line unicorn/consistent-function-scoping -- keep for readability
```
Comment (matthieu-crouzet) [inline packages/@o3r/extractors/schematics/cms-adapter/index.ts:66]: Could be extracted

File: packages/@o3r/extractors/src/core/comparator/package-managers-extractors/yarn2-file-extractor.helper.ts [hunk 20]
```diff
@@ -134,6 +134,7 @@ async function getProject(cwd = process.cwd()) {
   if (!configuration.projectCwd) {
     throw new O3rCliError(`No project found from ${cwd}`);
   }
+  // eslint-disable-next-line unicorn/no-array-method-this-argument -- null checked above
```
Comment (matthieu-crouzet) [inline packages/@o3r/extractors/src/core/comparator/package-managers-extractors/yarn2-file-extractor.helper.ts:137]:
```suggestion
  // eslint-disable-next-line unicorn/no-array-method-this-argument -- false positive `Project.find` is not an array method
```

File: packages/@o3r/extractors/src/utils/config-doc.ts [hunk 21]
```diff
@@ -85,7 +85,7 @@ export function getWidgetInformationFromDocComment(docText: string): ConfigPrope
       const valueText = text.slice(firstSpaceIndex + 1);
 
       try {
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- surrounded by try catch
```
Comment (matthieu-crouzet) [inline packages/@o3r/extractors/src/utils/config-doc.ts:88]: this rule is set as warn we can remove this line 
```suggestion
```

File: packages/@o3r/extractors/src/utils/validator.ts [hunk 22]
```diff
@@ -18,6 +18,7 @@ export function validateJson(jsonObject: unknown, schema: Record<string, unknown
     if (strictMode) {
       throw new Error(aggregatedError);
     }
+    // eslint-disable-next-line no-console -- needed to warn the user
```
Comment (matthieu-crouzet) [inline packages/@o3r/extractors/src/utils/validator.ts:21]:
```suggestion
    // eslint-disable-next-line no-console -- no other logger available
```

File: packages/@o3r/forms/src/core/helpers.ts [hunk 23]
```diff
@@ -62,7 +60,8 @@ export function getFlatControlErrors(form: AbstractControl) {
    * @note It filters out the 'customErrors'
    * @param errors Validation errors in a control
    */
-  const _getErrorList = (errors: ValidationErrors): FlatError[] => {
+  // eslint-disable-next-line unicorn/consistent-function-scoping -- keep for readability
```
Comment (matthieu-crouzet) [inline packages/@o3r/forms/src/core/helpers.ts:63]: could be extracted

File: packages/@o3r/forms/src/validators/number/max.directive.ts [hunk 24]
```diff
@@ -28,7 +28,7 @@ import {
       multi: true
     }
   ],
-  // eslint-disable-next-line @typescript-eslint/naming-convention
+  // eslint-disable-next-line @typescript-eslint/naming-convention -- convention required by HTML
```
Comment (matthieu-crouzet) [inline packages/@o3r/forms/src/validators/number/max.directive.ts:31]:
```suggestion
  // eslint-disable-next-line @typescript-eslint/naming-convention -- convention required by Angular
```

File: packages/@o3r/forms/src/validators/number/min.directive.ts [hunk 25]
```diff
@@ -28,7 +28,7 @@ import {
       multi: true
     }
   ],
-  // eslint-disable-next-line @typescript-eslint/naming-convention
+  // eslint-disable-next-line @typescript-eslint/naming-convention -- convention required by HTML
```
Comment (matthieu-crouzet) [inline packages/@o3r/forms/src/validators/number/min.directive.ts:31]:
```suggestion
  // eslint-disable-next-line @typescript-eslint/naming-convention -- convention required by Angular
```

File: tools/github-actions/audit/packaged-action/index.js
Comment (matthieu-crouzet) [inline tools/github-actions/audit/packaged-action/index.js:1]: You should not have changes in the folder `tools`

File: packages/@o3r/design/builders/generate-jsonschema/index.ts [hunk 26]
```diff
@@ -24,6 +24,7 @@ import type {
 } from './schema';
 
 const createBuilderWithMetricsIfInstalled: BuilderWrapper = (builderFn) => async (opts, ctx) => {
+  // eslint-disable-next-line unicorn/consistent-function-scoping -- keep for readability
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/builders/generate-jsonschema/index.ts:27]: Could be extracted like in `packages/@o3r/design/builders/generate-css/index.ts`

File: packages/@o3r/design/schematics/generate-css/index.ts [hunk 27]
```diff
@@ -24,7 +24,9 @@ import type {
 function generateCssFn(options: GenerateCssSchematicsSchema): Rule {
   return async (tree, context) => {
     const writeFile = (filePath: string, content: string) => tree.exists(filePath) ? tree.overwrite(filePath, content) : tree.create(filePath, content);
+    // eslint-disable-next-line @typescript-eslint/unbound-method -- method is not called here, we use the reference for the renderer options
     const readFile = tree.readText;
+    // eslint-disable-next-line @typescript-eslint/unbound-method -- method is not called here, we use the reference for the renderer options
     const existsFile = tree.exists;
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/schematics/generate-css/index.ts:30]:
```suggestion
    const readFile = tree.readText.bind(tree);
    const existsFile = tree.exists.bind(tree);
```

File: packages/@o3r/design/src/core/design-token/design-token-specification.interface.ts [hunk 28]
```diff
@@ -134,8 +134,7 @@ export interface DesignTokenTypeStrokeStyle<T extends DesignTokenTypeStrokeStyle
 type DesignTokenTypeBorderValue = {
   color: string;
   width: string;
-  // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
-  style: string | DesignTokenTypeStrokeStyleValue;
+  style: DesignTokenTypeStrokeStyleValue;
```
Comment (matthieu-crouzet) [inline packages/@o3r/design/src/core/design-token/design-token-specification.interface.ts:137]:
```suggestion
  style: Exclude<DesignTokenTypeStrokeStyleValue, string> | string;
```

File: eslint.shared.config.mjs [hunk 29]
```diff
@@ -73,7 +73,7 @@ export default [
   },
   {
     name: '@o3r/setup-jest',
-    files: ['**/setup-jest.ts'],
+    files: ['**/setup-jest.ts', '**/setup-jest.builders.ts'],
```
Comment (kpanot) [inline eslint.shared.config.mjs:76]:
```suggestion
    files: ['**/setup-jest.ts', '**/setup-jest.*.ts'],
```

to be prepared :)

File: packages/@o3r/design/cli/build-design-token.cli.cts [hunk 30]
```diff
@@ -48,6 +48,7 @@ void (async () => {
   )).reduce<DesignTokenVariableSet>((acc, { file, parsed }) => {
     parsed.forEach((variable, key) => {
       if (acc.has(key)) {
+        // eslint-disable-next-line no-console -- no logger available
```
Comment (kpanot) [inline packages/@o3r/design/cli/build-design-token.cli.cts:51]: should not be disable for all files matching `*.cli.cts` or `cli/*.cts` ?

File: packages/@o3r/design/schematics/extract-token/index.ts [hunk 31]
```diff
@@ -46,7 +46,7 @@ function extractTokenFn(options: ExtractTokenSchematicsSchema): Rule {
 
   return async (tree, context) => {
     try {
-      // eslint-disable-next-line @typescript-eslint/naming-convention
+      // eslint-disable-next-line @typescript-eslint/naming-convention -- false positive when using dynamic import
```
Comment (kpanot) [inline packages/@o3r/design/schematics/extract-token/index.ts:49]: That's a bit annoying actually :S.
Do we have this issue on all Class dynamically imported ? Is it possible to have exception for this case (which seems quite common) ?

File: packages/@o3r/design/src/core/design-token/design-token-specification.interface.ts (hunk 28 above)
Comment (kpanot) [reply packages/@o3r/design/src/core/design-token/design-token-specification.interface.ts:137]: We discussed and `string |` was actually a mistake

File: packages/@o3r/dev-tools/eslint.local.config.mjs [hunk 32]
```diff
@@ -23,5 +23,11 @@ export default [
         ]
       }
     }
+  },
+  {
+    name: '@o3r/dev-tools/override',
+    rules: {
+      'unicorn/no-process-exit': 'off'
```
Comment (kpanot) [inline packages/@o3r/dev-tools/eslint.local.config.mjs:30]: This is the case for CLIs only (and maybe all the CLIs of the repo?)

File: packages/@o3r/dev-tools/src/cli/artifact-retriever.ts [hunk 33]
```diff
@@ -202,11 +201,15 @@ function getLatestVersion(packages: Record<string, any>[], artifactName: string)
...
-      const res = await request.get(`https://feeds.dev.azure.com/${opts.organization as string}/${opts.project as string}/_apis/packaging/feeds/${opts.feed as string}/packages?api-version=6.0-preview.1`, options).promise();
+      const res = await request.get(
+        `https://feeds.dev.azure.com/${opts.organization as string}/${opts.project as string}/_apis/packaging/feeds/${opts.feed as string}/packages?api-version=6.0-preview.1`,
+        options
+      ).promise();
       version = getLatestVersion(JSON.parse(res).value, name);
     }
     logger.info(`Searching for ${name}@${version}`);
 
+    // eslint-disable-next-line @stylistic/max-len -- keep url on same line
```
Comment (kpanot) [inline packages/@o3r/dev-tools/src/cli/artifact-retriever.ts:212]: I would, personally, splitted it in multiline with `+` joiner

File: packages/@o3r/eslint-config-otter/schematics/ng-add/index.ts [hunk 34]
```diff
@@ -55,7 +55,7 @@ function ngAddFn(options: NgAddSchematicsSchema): Rule {
     const depsInfo = getO3rPeerDeps(path.resolve(__dirname, '..', '..', 'package.json'), true, /^@(?:o3r|ama-sdk|eslint-)/);
     const workspaceProject = options.projectName ? getWorkspaceConfig(tree)?.projects[options.projectName] : undefined;
     const linterSchematicsFolder = __dirname;
-    // eslint-disable-next-line @typescript-eslint/naming-convention
+    // eslint-disable-next-line @typescript-eslint/naming-convention -- false positive when using dynamic import
     const { NodeDependencyType } = await import('@schematics/angular/utility/dependencies');
```
Comment (kpanot) [inline packages/@o3r/eslint-config-otter/schematics/ng-add/index.ts:59]: Same issue as Class.
Maybe we should find a proper workaround for this rule

File: packages/@o3r/eslint-config/eslint.local.config.mjs [hunk 35]
```diff
@@ -17,12 +17,24 @@ export default [
...
           'tsconfig.spec.json'
         ]
       }
     }
+  },
+  {
+    name: '@o3r/eslint-config/projects/commonjs',
+    files: ['**/*.cjs'],
+    languageOptions: {
+      sourceType: 'commonjs',
```
Comment (kpanot) [inline packages/@o3r/eslint-config/eslint.local.config.mjs:31]: this is potentially global at workspace level (for `.cjs`).
A similar rule can be set for `.mjs`

File: packages/@o3r/design/src/core/design-token/design-token-specification.interface.ts (hunk 28 above)
Comment (matthieu-crouzet) [reply packages/@o3r/design/src/core/design-token/design-token-specification.interface.ts:137]: maybe should be done in an another PR then 
else we may think at some point that we lost this during eslint fixes

File: packages/@o3r/design/cli/build-design-token.cli.cts (hunk 30 above)
Comment (matthieu-crouzet) [reply packages/@o3r/design/cli/build-design-token.cli.cts:51]: depends sometimes we have a winstom logger sometimes not...

File: packages/@o3r/design/schematics/extract-token/index.ts (hunk 31 above)
Comment (matthieu-crouzet) [reply packages/@o3r/design/schematics/extract-token/index.ts:49]: We can have a look if it's possible to detect them with this rule 
this rule seems to accept a lot of configurations

File: packages/@o3r/dev-tools/eslint.local.config.mjs (hunk 32 above)
Comment (matthieu-crouzet) [reply packages/@o3r/dev-tools/eslint.local.config.mjs:30]: This package are only dev tools for CLI no ? and it is deprecated we will remove it in v12

File: packages/@o3r/design/src/core/design-token/design-token-specification.interface.ts (hunk 28 above)
Comment (kpanot) [reply packages/@o3r/design/src/core/design-token/design-token-specification.interface.ts:137]: PR done on #2362

File: packages/@o3r/dev-tools/src/cli/aql-artifact-cleaner.ts [hunk 36]
```diff
@@ -63,18 +63,12 @@ if (!opts.basicAuth && !opts.apiKey) {
...
-    // eslint-disable-next-line @typescript-eslint/naming-convention
-    'X-JFrog-Art-Api': (opts.apiKey as string)
-  };
-}
+const authHeader: Headers = {
+  ...(
+    // eslint-disable-next-line @typescript-eslint/naming-convention -- naming convention required by HTTP
+    opts.basicAuth ? { Authorization: 'Basic ' + (opts.basicAuth as string) } : { 'X-JFrog-Art-Api': (opts.apiKey as string) }
+  )
+};
```
Comment (fpaul-1A) [inline packages/@o3r/dev-tools/src/cli/aql-artifact-cleaner.ts:71]:
```suggestion
const authHeader: Headers = opts.basicAuth
  ? { Authorization: 'Basic ' + (opts.basicAuth as string) }
  : { 'X-JFrog-Art-Api': (opts.apiKey as string) };
```

File: packages/@o3r/dynamic-content/src/services/request-parameters/request-parameters.config.ts [hunk 37]
```diff
@@ -42,6 +42,6 @@ export interface RequestParametersConfig {
 export const defaultRequestParametersConfig: RequestParametersConfig = {
   storage: (typeof window === 'undefined') ? undefined : window.sessionStorage,
   strategy: StorageStrategy.Rehydrate,
-  queryParamsValue: typeof document !== 'undefined' && document.body && document.body.dataset && document.body.dataset.query || '{}',
-  postParamsValue: typeof document !== 'undefined' && document.body && document.body.dataset && document.body.dataset.post || '{}'
+  queryParamsValue: (typeof document !== 'undefined' && document.body && document.body.dataset && document.body.dataset.query) || '{}',
+  postParamsValue: (typeof document !== 'undefined' && document.body && document.body.dataset && document.body.dataset.post) || '{}'
```
Comment (fpaul-1A) [inline packages/@o3r/dynamic-content/src/services/request-parameters/request-parameters.config.ts:46]:
```suggestion
  queryParamsValue: (typeof document !== 'undefined' && document.body?.dataset?.query) || '{}',
  postParamsValue: (typeof document !== 'undefined' && document.body?.dataset?.post) || '{}'
```

File: packages/@o3r/extractors/src/utils/localization.ts [hunk 38]
```diff
@@ -18,7 +18,7 @@ export function getLocalizationFileFromAngularElement(node: ts.ClassDeclaration)
         !ts.isDecorator(decorator)
         || !ts.isCallExpression(decorator.expression)
         || !ts.isIdentifier(decorator.expression.expression)
-        || decorator.expression.expression.escapedText !== localizationDecoratorName
+        || decorator.expression.expression.escapedText.toString() !== localizationDecoratorName
```
Comment (fpaul-1A) [inline packages/@o3r/extractors/src/utils/localization.ts:21]: I thought the rule was preventing us from using `.toString()` (@typescript-eslint/no-base-to-string), is it ok here?

File: packages/@o3r/design/schematics/ng-add/index.ts [hunk 39]
```diff
@@ -24,22 +24,14 @@ export function ngAddFn(options: NgAddSchematicsSchema): Rule {
   /* ng add rules */
   return async (tree) => {
     const { getPackageInstallConfig, setupDependencies, setupSchematicsParamsForProject } = await import('@o3r/schematics');
+    const defaultSchematicOptions = Object.fromEntries([
+      '@o3r/core:component',
+      '@o3r/core:component-presenter',
+      '*:*'
+    ].map((name) => [name, { useOtterDesignToken: true }]));
```
Comment (cpaulve-1A) [inline packages/@o3r/design/schematics/ng-add/index.ts:31]: Is it really equivalent ? We have a two entry list here when it was a key value object before?

Comment (matthieu-crouzet) [reply packages/@o3r/design/schematics/ng-add/index.ts:31]: image
Object.fromEntries return an object

File: packages/@o3r/extractors/src/utils/localization.ts (hunk 38 above)
Comment (matthieu-crouzet) [reply packages/@o3r/extractors/src/utils/localization.ts:21]: Here it's okay as `escapedText` expose explicitly a `toString` method

File: packages/@o3r/design/builders/generate-css/index.ts [hunk 40]
```diff
@@ -80,7 +80,7 @@ export default createBuilder<GenerateCssSchematicsSchema>(createBuilderWithMetri
   });
   const writeFileWithLogger: typeof writeFile = async (file, ...args) => {
     const res = await writeFile(file, ...args);
-    // eslint-disable-next-line @typescript-eslint/no-base-to-string
+    // eslint-disable-next-line @typescript-eslint/no-base-to-string -- convert file to string for the logger
     logger.info(`Updated ${file.toString()} with Design Token content.`);
```
[the remark this answers is not in the document]
Comment (matthieu-crouzet) [reply packages/@o3r/design/builders/generate-css/index.ts:84]: I think the type of writeFileWithLogger is too permissive compared to what we really need

this should be enough
```ts
const writeFileWithLogger: DesignTokenRendererOptions['writeFile'] = async (file, content) => {
    const res = await writeFile(file, content);
    logger.info(`Updated ${file} with Design Token content.`);
    return res;
  };

File: packages/@o3r/design/schematics/ng-add/index.ts (hunk 39 above)
Comment (sdo-1A) [reply packages/@o3r/design/schematics/ng-add/index.ts:31]: https://github.com/AmadeusITGroup/otter/pull/2384

File: packages/@o3r/eslint-config/eslint.local.config.mjs (hunk 35 above)
Comment (matthieu-crouzet) [reply packages/@o3r/eslint-config/eslint.local.config.mjs:31]: A global rule is set, here it is needed to linked the tsconfig.build

File: tools/github-actions/audit/packaged-action/index.js
Comment (fpaul-1A) [inline tools/github-actions/audit/packaged-action/index.js:1]: it's weird to see the diff in this PR, it should have been already merged with the fixes of `tools/github-actions`

Comment (matthieu-crouzet) [reply tools/github-actions/audit/packaged-action/index.js:1]: done

File: packages/@o3r/design/builders/generate-css/index.ts [hunk 41]
```diff
@@ -43,8 +43,10 @@ import type {
   GenerateCssSchematicsSchema
 } from './schema';
 
+const noopBuilder: BuilderWrapper = (fn) => fn;
+
 const createBuilderWithMetricsIfInstalled: BuilderWrapper = (builderFn) => async (opts, ctx) => {
-  let wrapper: BuilderWrapper = (fn) => fn;
+  let wrapper: BuilderWrapper = noopBuilder;
```
Comment (mrednic-1A) [inline packages/@o3r/design/builders/generate-css/index.ts:49]: you should not need the type explicit here.

**Review by mrednic-1A — APPROVED** [review]

**Pull request merged**